src/formState.ts

```diff
@@ -346,13 +347,16 @@ function newObjectState<T, P = any>(
     },
 
     get readOnly(): boolean {
-      return this._readOnly || this._considerReadOnly();
+      return (
+        this._readOnly ||
+        this._considerReadOnly() ||
+        (parentState && parentState().readOnly) ||
+        parentListState?.readOnly
+      );
```

Comment:
This `readOnly` is for object's, i.e. either top-level objects `author.readOnly` or child objects witin a list `author.books.rows[0].readOnly`.